pyccel/codegen/printing/fcode.py

```diff
@@ -452,28 +455,151 @@ def _print_Import(self, expr):
+                args_format.append(arg_format)
+                args.append(arg)
+            elif f.dtype is NativeString() and f != expr.expr[-1]:
+                args_format.append('A')
                 args.append(self._print(f))
+            elif isinstance(f.rank, int) and f.rank > 0:
+                if args_format:
+                    code += formatted_args_to_print(args_format, args, self._print_LiteralString(LiteralString(sep)))
+                    args_format = []
+                    args = []
+                for_index = Variable(NativeInteger(), name=self.parser.get_new_name('i'))
+                self._additional_declare.append(for_index)
+                max_index = PyccelMinus(f.shape[0], LiteralInteger(1), simplify=True)
+                for_range = PythonRange(max_index)
+                print_body = [FunctionCallArgument(f[for_index])]
+                if f.rank == 1:
+                    print_body.append(space_end)
+
+                for_body = [PythonPrint(print_body)]
+                for_loop = For(for_index, for_range, for_body)
+                for_end = FunctionCallArgument(LiteralString(']' + end if i == len(orig_args) - 1 else ']'),
+                                               keyword='end')
+
+                body = CodeBlock([PythonPrint([FunctionCallArgument(LiteralString('[')), empty_end]),
+                                  for_loop,
+                                  PythonPrint([FunctionCallArgument(f[max_index]), for_end])],
+                                 unravelled=True)
+                code += self._print(body)
+            else:
+                arg_format, arg = self.get_print_format_and_arg(f)
+                args_format.append(arg_format)
+                args.append(arg)
+        if args_format:
+            code += formatted_args_to_print(args_format, args, end)
+        return code
 
-        code = ', '.join(['print *', *args])
-        return self._get_statement(code) + '\n'
+    def get_print_format_and_arg(self,var):
+        type_to_format = {('real'): 'F0.12',
+                          ('complex'): '"(",F0.12," + ",F0.12,")"',
+                          ('integer'): 'I0',
+                          ('logical'): 'A',
+                          ('string'): 'A'}
+        var_type = self._print(var.dtype)
+        if var_type not in type_to_format:
+            if self._print(var.dtype).find("character") != -1:
+                return 'A', self._print(var)
+            return '*', self._print(var)
+
+        try:
+            arg_format = type_to_format[var_type]
+        except KeyError:
+            errors.report("{} type is not supported currently".format(var.dtype), severity='fatal')
+
+        if var.dtype is NativeComplex():
+            arg = '{}, {}'.format(self._print(NumpyReal(var)), self._print(NumpyImag(var)))
+        elif var.dtype is NativeBool():
+            arg = 'merge("True ", "False", {})'.format(self._print(var))
```

Comment:
Perhaps we could also handle `LiteralTrue()` and `LiteralFalse()` explicitly, it seems strange to write:
```
merge("True ", "False", .True.)
```